src/auto-evo/Miche.cs

```diff
@@ -246,4 +266,21 @@ public override int GetHashCode()
         return Pressure.GetHashCode() * 131 ^ parentHash * 587 ^
             (Occupant == null ? 17 : Occupant.GetHashCode()) * 5171;
     }
+
+    public class InsertWorkingMemory
```

Comment:
Documenting this class and making this reusable for new miches would help a lot in this class' usage.